bluepysnap/config.py

```diff
@@ -163,11 +164,51 @@ def to_dict(self):
 class CircuitConfig(Config):
     """Handle CircuitConfig."""
 
+    def __init__(self, *args):
+        """Initializes circuit config."""
+        super().__init__(*args)
+        self._populations = self._resolve_population_configs()
+
     @classmethod
     def from_config(cls, config_path):
         """Instantiate the config class from circuit configuration."""
         return cls(config_path, libsonata.CircuitConfig)
 
+    @property
+    def populations(self):
+        """Access population configs."""
+        return self._populations
+
+    # NOTE: These are parsed here to keep Parser returning the exact parsed configuration.
+    def _resolve_population_configs(self):
+        """Resolves population configs for the node and edge populations."""
+        config = self.to_dict()
+        networks = config.get("networks") or {}
+        components = config.get("components") or {}
+
+        def resolve_network_populations(element_type):
+            populations = {}
+            type_file_key = f"{element_type}_file"
+            element_dicts = networks.get(element_type) or []
+
+            for elem_dict in element_dicts:
+                elem_pops = elem_dict.get("populations") or {}
+
+                for pop, cfg in elem_pops.items():
+                    populations[pop] = copy.deepcopy(components)
+                    populations[pop].update(cfg or {})
+
+                    # add h5 filepath for simpler access
+                    if type_file_key in elem_dict:
+                        populations[pop][type_file_key] = elem_dict[type_file_key]
+
+            return populations
+
+        return {
+            "node": resolve_network_populations("nodes"),
+            "edge": resolve_network_populations("edges"),
```

Comment:
Minor, keeping the plural names "nodes" and "edges" seems more natural to me, but it can be subjective.

Reply:
Good point, I'll add a few tests to make sure that the components and population config is done in the correct order etc., so might as well change this to be uniform with everything else.